store/sqlstore/channel_store.go

```diff
@@ -2518,6 +2526,51 @@ func (s SqlChannelStore) UpdateLastViewedAt(channelIds []string, userId string)
 	return times, nil
 }
 
+func (s SqlChannelStore) CountUrgentPostsAfter(channelId string, timestamp int64, userId string) (int, error) {
+	joinLeavePostTypes := []string{
+		// These types correspond to the ones checked by Post.IsJoinLeaveMessage
+		model.PostTypeJoinLeave,
+		model.PostTypeAddRemove,
+		model.PostTypeJoinChannel,
+		model.PostTypeLeaveChannel,
+		model.PostTypeJoinTeam,
+		model.PostTypeLeaveTeam,
+		model.PostTypeAddToChannel,
+		model.PostTypeRemoveFromChannel,
+		model.PostTypeAddToTeam,
+		model.PostTypeRemoveFromTeam,
+	}
+
+	query := s.getQueryBuilder().
+		Select("count(*)").
+		From("Posts").
+		Join("PostsPriority ON Posts.Id = PostsPriority.PostId").
```

Comment:
It shouldn't matter from the database perspective since it ought to optimize to the same result, but would it make sense to write this as a query `FROM PostsPriority` and `JOIN Posts` to clarify the source of truth at play?